From 46813d91d4d5aa9cb62f58b7e3c9281eeb82f13d Mon Sep 17 00:00:00 2001
From: CI Assistant <yossi.ilkanaev@intel.com>
Date: Thu, 20 Dec 2018 20:53:15 +0100
Subject: [PATCH] Merge pull request #335 in SW_UGW/linux from
 bugfix/UGW_SW-22476-no-video-playback-on-main-ap-8.x to master

* commit '6d8cdbea6b6a2ebd4509a271ee400ef91b398213':
  UGW_SW-22476 : Fixed string compare issue in Linux net_device names (e.g. wlan0.0 with wlan0) and Linux module names.
---
 net/mcast_helper/mcast_helper_reg.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/mcast_helper/mcast_helper_reg.c b/net/mcast_helper/mcast_helper_reg.c
index 2b7cfac3ff1b..bb0cd5c394d1 100644
--- a/net/mcast_helper/mcast_helper_reg.c
+++ b/net/mcast_helper/mcast_helper_reg.c
@@ -172,8 +172,8 @@ static int mcast_helper_dereg_callback(struct net_device *netdev,
 		mc_callback_rec = list_entry(liter, MCAST_CALLBACK_t, list);
 		if (mc_callback_rec != NULL) {
 			if (mc_callback_rec->netDev->name != NULL) {
-				if (!strncmp (netdev->name, mc_callback_rec->netDev->name, strlen(mc_callback_rec->netDev->name))) {
-					if (!strncmp (modName->name, mc_callback_rec->modName->name, strlen(mc_callback_rec->netDev->name))) {
+				if (!strncmp (netdev->name, mc_callback_rec->netDev->name, IFNAMSIZ)) {
+					if (!strncmp (modName->name, mc_callback_rec->modName->name, MODULE_NAME_LEN)) {
 						list_del(&mc_callback_rec->list);
 						kfree(mc_callback_rec);
 						if ((flags & LTQ_MC_F_FW_RESET) == LTQ_MC_F_FW_RESET)
